plugins/scaffolder-backend/src/scaffolder/actions/builtin/bitbucket-cloud/bitbucketCloudRepoCreate.ts

```diff
@@ -0,0 +1,192 @@
+            title: 'Default Branch',
+            type: 'string',
+            description: `Sets the default branch on the repository. The default value is 'master'`,
+          },
+          sourcePath: {
+            title: 'Source Path',
+            description:
+              'Path within the workspace that will be used as the repository root. If omitted, the entire workspace will be published as the repository.',
+            type: 'string',
+          },
+          enablePipeline: {
+            title: 'Enable Bitbucket pipeline',
+            description:
+              'Enable Bitbucket pipelines on this repository. Default is False',
+            type: 'boolean',
+          },
+          token: {
+            title: 'Authentication Token',
+            type: 'string',
+            description:
+              'The token to use for authorization to BitBucket Cloud',
+          },
+        },
+      },
+      output: {
+        type: 'object',
+        properties: {
+          remoteUrl: {
+            title: 'A URL to the repository with the provider',
+            type: 'string',
+          },
+          repoContentsUrl: {
+            title: 'A URL to the root of the repository',
+            type: 'string',
+          },
+        },
+      },
+    },
+    async handler(ctx) {
+      const {
+        description,
+        defaultBranch = 'master',
+        enablePipeline = false,
+        repoVisibility = 'private',
+        token,
+        repoUrl,
+      } = ctx.input;
+
+      const {
+        apiBaseUrl,
+        authorization,
+        integrationConfig,
+        project,
+        repo,
+        workspace,
+      } = getBitbucketCloudConfig({ repoUrl, token, integrations });
+
+      // Use Bitbucket API to create a new repository
+      const { remoteUrl, repoContentsUrl } =
+        await bitbucketCloudCreateRepository({
+          authorization,
+          workspace: workspace || '',
+          project,
+          repo,
+          repoVisibility,
+          mainBranch: defaultBranch,
+          description,
+          apiBaseUrl,
+        });
+
+      const gitAuthorInfo = {
+        name: config.getOptionalString('scaffolder.defaultAuthor.name'),
+        email: config.getOptionalString('scaffolder.defaultAuthor.email'),
+      };
+
+      let auth;
+
+      if (ctx.input.token) {
+        auth = {
+          username: 'x-token-auth',
+          password: ctx.input.token,
+        };
+      } else {
+        if (
+          !integrationConfig.config.username ||
+          !integrationConfig.config.appPassword
+        ) {
+          throw new Error(
+            'Credentials for Bitbucket Cloud integration required for this action.',
+          );
+        }
+
+        auth = {
+          username: integrationConfig.config.username,
+          password: integrationConfig.config.appPassword,
+        };
+      }
+      // Use isomorphic-git to push workspace to new repository
+      await initRepoAndPush({
```

Comment:
this part is a bit questionable for me. Basically, this is just `publish:bitbucketCloud` and then adding support for `enable pipelines` could have been an option.

Maybe it is good to split "create repo" and "push to repo" here.

Reply:
+1 here. I believe thats the difference between `publish:github` and `github:repo:create`